### ISSUE TITLE
VAE Training - generalization and reconstruction capability

### ISSUE THREAD

**Issue by oahmet1** [NSAPH-Projects/pm-mortality-generative#1]
Current VAEs problems
- First 50 Epochs the loss decreases significantly, then for the next 120 Epochs only very little : 
https://github.com/NSAPH-Projects/pm-mortality-generative/blob/main/run_logs/pytorch_3904699.out

- It can not reconstruct data points well. One of the good recreations is the following :
[real image ](https://github.com/NSAPH-Projects/pm-mortality-generative/blob/main/preprocessing/dataset_samples/outcome_0/real_1.png) [reconstructed image](https://github.com/NSAPH-Projects/pm-mortality-generative/blob/main/preprocessing/reconstructed_samples/outcome_0/sample_1.png) 

- Sampling from the latent distribution and running the decoder creates images that are (I would assume) outside of the dataset distribution
[like this one](https://github.com/NSAPH-Projects/pm-mortality-generative/blob/main/preprocessing/reconstructed_samples/outcome_0/sample_0.png)

Figure out what is hindering current models capabilities. Is it one of the following  ? Size/ quality of the dataset, the VAE model, learning parameters ?
